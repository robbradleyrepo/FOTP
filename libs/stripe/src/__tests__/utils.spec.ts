import {
  fromPaymentRequestShippingAddress,
  toSmallestCurrencyUnit,
} from "../utils";

describe("fromPaymentRequestShippingAddress", () => {
  it("should format complete address details with a parsable name", () => {
    expect(
      fromPaymentRequestShippingAddress({
        addressLine: ["123 Fake St"],
        city: "Los Angeles",
        country: "US",
        phone: "937-901-5687",
        postalCode: "90017",
        recipient: "Joe Q Bloggs",
        region: "California",
      })
    ).toMatchSnapshot();
  });

  // The `parse-full-name` library doesn't handle middle initials that are also
  // conjunctions - see https://github.com/RateGravity/parse-full-name/issues/7
  it("should format complete address details with a parsable name that included the initials that are also conjunctions", () => {
    expect(
      fromPaymentRequestShippingAddress({
        addressLine: ["123 Fake St"],
        city: "Los Angeles",
        country: "US",
        phone: "937-901-5687",
        postalCode: "90017",
        recipient: "Joe and Jane Bloggs",
        region: "California",
      })
    ).toMatchSnapshot();

    expect(
      fromPaymentRequestShippingAddress({
        addressLine: ["123 Fake St"],
        city: "Los Angeles",
        country: "US",
        phone: "937-901-5687",
        postalCode: "90017",
        recipient: "Joe ET Bloggs",
        region: "California",
      })
    ).toMatchSnapshot();

    expect(
      fromPaymentRequestShippingAddress({
        addressLine: ["123 Fake St"],
        city: "Los Angeles",
        country: "US",
        phone: "937-901-5687",
        postalCode: "90017",
        recipient: "Joe E Bloggs",
        region: "California",
      })
    ).toMatchSnapshot();

    expect(
      fromPaymentRequestShippingAddress({
        addressLine: ["123 Fake St"],
        city: "Los Angeles",
        country: "US",
        phone: "937-901-5687",
        postalCode: "90017",
        recipient: "Joe Y Bloggs",
        region: "California",
      })
    ).toMatchSnapshot();
  });

  it("should format complete address details with an unparsable or incomplete name", () => {
    const details = {
      addressLine: ["123 Fake St"],
      city: "Los Angeles",
      country: "US",
      phone: "937-901-5687",
      postalCode: "90017",
      region: "California",
    };

    expect(
      fromPaymentRequestShippingAddress({
        ...details,
        recipient: "Madonna",
      })
    ).toMatchSnapshot();

    expect(
      fromPaymentRequestShippingAddress({
        ...details,
        recipient: "Ms Ciccone",
      })
    ).toMatchSnapshot();

    expect(
      fromPaymentRequestShippingAddress({
        ...details,
        recipient:
          "Adolph Blaine Charles David Earl Frederick Gerald Hubert Irvin John Kenneth Lloyd Martin Nero Oliver Paul Quincy Randolph Sherman Thomas Uncas Victor William Xerxes Yancy Zeus Wolfeschlegel­steinhausen­bergerdorff­welche­vor­altern­waren­gewissenhaft­schafers­wessen­schafe­waren­wohl­gepflege­und­sorgfaltigkeit­beschutzen­vor­angreifen­durch­ihr­raubgierig­feinde­welche­vor­altern­zwolfhundert­tausend­jahres­voran­die­erscheinen­von­der­erste­erdemensch­der­raumschiff­genacht­mit­tungstein­und­sieben­iridium­elektrisch­motors­gebrauch­licht­als­sein­ursprung­von­kraft­gestart­sein­lange­fahrt­hinzwischen­sternartig­raum­auf­der­suchen­nachbarschaft­der­stern­welche­gehabt­bewohnbar­planeten­kreise­drehen­sich­und­wohin­der­neue­rasse­von­verstandig­menschlichkeit­konnte­fortpflanzen­und­sich­erfreuen­an­lebenslanglich­freude­und­ruhe­mit­nicht­ein­furcht­vor­angreifen­vor­anderer­intelligent­geschopfs­von­hinzwischen­sternartig­raum Sr.",
      })
    ).toMatchSnapshot();
  });

  it("should format partial address details when explicitly allowed", () => {
    expect(
      fromPaymentRequestShippingAddress(
        {
          city: "Los Angeles",
          country: "US",
          postalCode: "90017",
          region: "California",
        },
        true
      )
    ).toMatchSnapshot();
  });

  it("should throw an error on partial address details if not explicitly allowed", () => {
    expect(() =>
      fromPaymentRequestShippingAddress({
        city: "Los Angeles",
        country: "US",
        postalCode: "90017",
        region: "California",
      })
    ).toThrowError("Missing address fields");
  });
});

describe("toSmallestCurrencyUnit", () => {
  it("should convert decimal currencies in major units into the corresponding minor unit", () => {
    expect(
      toSmallestCurrencyUnit({ amount: "1.23", currencyCode: "AUD" })
    ).toBe(123);
    expect(
      toSmallestCurrencyUnit({ amount: "0.123", currencyCode: "GBP" })
    ).toBe(12);
    expect(toSmallestCurrencyUnit({ amount: "123", currencyCode: "USD" })).toBe(
      12300
    );
  });

  it("should parse zero-decimal currencies without multiplying them", () => {
    expect(
      toSmallestCurrencyUnit({ amount: "1.23", currencyCode: "BIF" })
    ).toBe(1);
    expect(
      toSmallestCurrencyUnit({ amount: "0.123", currencyCode: "JPY" })
    ).toBe(0);
    expect(toSmallestCurrencyUnit({ amount: "123", currencyCode: "XPF" })).toBe(
      123
    );
  });
});
