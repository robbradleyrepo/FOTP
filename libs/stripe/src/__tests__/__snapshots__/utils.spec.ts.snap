// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fromPaymentRequestShippingAddress should format complete address details with a parsable name 1`] = `
Object {
  "address1": "123 Fake St",
  "address2": "",
  "city": "Los Angeles",
  "company": "",
  "countryCode": "US",
  "firstName": "Joe",
  "lastName": "Bloggs",
  "phone": "937-901-5687",
  "province": "California",
  "zip": "90017",
}
`;

exports[`fromPaymentRequestShippingAddress should format complete address details with a parsable name that included the initials that are also conjunctions 1`] = `
Object {
  "address1": "123 Fake St",
  "address2": "",
  "city": "Los Angeles",
  "company": "",
  "countryCode": "US",
  "firstName": "Joe and Jane",
  "lastName": "Bloggs",
  "phone": "937-901-5687",
  "province": "California",
  "zip": "90017",
}
`;

exports[`fromPaymentRequestShippingAddress should format complete address details with a parsable name that included the initials that are also conjunctions 2`] = `
Object {
  "address1": "123 Fake St",
  "address2": "",
  "city": "Los Angeles",
  "company": "",
  "countryCode": "US",
  "firstName": "Joe",
  "lastName": "Bloggs",
  "phone": "937-901-5687",
  "province": "California",
  "zip": "90017",
}
`;

exports[`fromPaymentRequestShippingAddress should format complete address details with a parsable name that included the initials that are also conjunctions 3`] = `
Object {
  "address1": "123 Fake St",
  "address2": "",
  "city": "Los Angeles",
  "company": "",
  "countryCode": "US",
  "firstName": "Joe",
  "lastName": "Bloggs",
  "phone": "937-901-5687",
  "province": "California",
  "zip": "90017",
}
`;

exports[`fromPaymentRequestShippingAddress should format complete address details with a parsable name that included the initials that are also conjunctions 4`] = `
Object {
  "address1": "123 Fake St",
  "address2": "",
  "city": "Los Angeles",
  "company": "",
  "countryCode": "US",
  "firstName": "Joe",
  "lastName": "Bloggs",
  "phone": "937-901-5687",
  "province": "California",
  "zip": "90017",
}
`;

exports[`fromPaymentRequestShippingAddress should format complete address details with an unparsable or incomplete name 1`] = `
Object {
  "address1": "123 Fake St",
  "address2": "",
  "city": "Los Angeles",
  "company": "",
  "countryCode": "US",
  "firstName": "",
  "lastName": "Madonna",
  "phone": "937-901-5687",
  "province": "California",
  "zip": "90017",
}
`;

exports[`fromPaymentRequestShippingAddress should format complete address details with an unparsable or incomplete name 2`] = `
Object {
  "address1": "123 Fake St",
  "address2": "",
  "city": "Los Angeles",
  "company": "",
  "countryCode": "US",
  "firstName": "",
  "lastName": "Ms Ciccone",
  "phone": "937-901-5687",
  "province": "California",
  "zip": "90017",
}
`;

exports[`fromPaymentRequestShippingAddress should format complete address details with an unparsable or incomplete name 3`] = `
Object {
  "address1": "123 Fake St",
  "address2": "",
  "city": "Los Angeles",
  "company": "",
  "countryCode": "US",
  "firstName": "",
  "lastName": "Adolph Blaine Charles David Earl Frederick Gerald Hubert Irvin John Kenneth Lloyd Martin Nero Oliver Paul Quincy Randolph Sherman Thomas Uncas Victor William Xerxes Yancy Zeus Wolfeschlegel­steinhausen­bergerdorff­welche­vor­altern­waren­gewissenhaft­schafers­wessen­schafe­waren­wohl­gepflege­und­sorgfaltigkeit­beschutzen­vor­angreifen­durch­ihr­raubgierig­feinde­welche­vor­altern­zwolfhundert­tausend­jahres­voran­die­erscheinen­von­der­erste­erdemensch­der­raumschiff­genacht­mit­tungstein­und­sieben­iridium­elektrisch­motors­gebrauch­licht­als­sein­ursprung­von­kraft­gestart­sein­lange­fahrt­hinzwischen­sternartig­raum­auf­der­suchen­nachbarschaft­der­stern­welche­gehabt­bewohnbar­planeten­kreise­drehen­sich­und­wohin­der­neue­rasse­von­verstandig­menschlichkeit­konnte­fortpflanzen­und­sich­erfreuen­an­lebenslanglich­freude­und­ruhe­mit­nicht­ein­furcht­vor­angreifen­vor­anderer­intelligent­geschopfs­von­hinzwischen­sternartig­raum Sr.",
  "phone": "937-901-5687",
  "province": "California",
  "zip": "90017",
}
`;

exports[`fromPaymentRequestShippingAddress should format partial address details when explicitly allowed 1`] = `
Object {
  "address1": undefined,
  "address2": "",
  "city": "Los Angeles",
  "company": "",
  "countryCode": "US",
  "phone": "",
  "province": "California",
  "zip": "90017",
}
`;
